.github/workflows/node-ci.yml

```diff
@@ -0,0 +1,151 @@
+name: node-ci
+
+on:
+  workflow_dispatch:
+  push:
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - "platform/android/**"
+      - "platform/ios/**"
+      - ".gitignore"
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - "platform/android/**"
+      - "platform/ios/**"
+      - ".gitignore"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os:
+          - macos-11.0
+          - macos-10.15
+          - ubuntu-20.04
+          - ubuntu-18.04
+    runs-on: ${{ matrix.os }}
+    env:
+      BUILDTYPE: Debug
+
+    defaults:
+      run:
+        working-directory: ./
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Cache homebrew
+        uses: actions/cache@v2
+        if: runner.os == 'macOS'
+        env:
+          cache-name: homebrew-v1
+        with:
+          path: /usr/local/Cellar
+          key: ${{ env.cache-name }}-${{ runner.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}-${{ github.head_ref }}
+          restore-keys: |
+            ${{ env.cache-name }}-${{ runner.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}
+            ${{ env.cache-name }}-${{ runner.os }}-${{ github.job }}-${{ github.ref }}
+            ${{ env.cache-name }}-${{ runner.os }}-${{ github.job }}
+
+
+      - name: Install dependencies
+        if: runner.os == 'macOS'
+        env:
+          HOMEBREW_NO_AUTO_UPDATE: 1
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+        run: |
+          brew list cmake || brew install cmake
+          brew list ccache || brew install ccache
+          brew list ninja || brew install ninja
+          brew list pkgconfig || brew install pkgconfig
+          brew list qt || brew install qt
```

Comment:
I know this was already asked, but why exactly is Qt needed?

Reply:
It's basically because everything is built as a single package. It's not a direct dependency of the `node` build, but it seems to be coupled together.

Reply:
Then I consider this a bug in the setup. Each platfom should build independently.